nixos/modules/programs/xonsh.nix

```diff
@@ -5,6 +5,10 @@
 let
 
   cfg = config.programs.xonsh;
+  package = cfg.package.overridePythonAttrs (old: {
+    python3 = pkgs.python3.withPackages (ps: cfg.xontribs);
+    propagatedBuildInputs = old.propagatedBuildInputs ++ cfg.xontribs;
```

Comment:
I'm not sure why `pkgs.xonsh`, the wrapper, has `overridePythonAttrs`. But don't use it here anyway, as it returns the unwrapped xonsh package, which is not able to launch outside a python env.

Reply:
It inherits `python3Packages.xonsh`'s `passthru`, which led to that.

I'm really not sure why this doesn't use `extraPackages`, the documentation even implies it does. Fixing now.

Reply:
Alright fixed it to handle that properly, with the caveat that it now bakes in the assumption that the package will have an `extraPackages` overridable argument. (Added a module argument that composes with xontribs as a compromise.)